CMakeLists.txt

```diff
@@ -632,13 +629,14 @@ endif()
 
 if(AVIF_CODEC_AOM_ENABLED)
     get_target_property(AOM_INTERFACE_LINK_LIBRARIES aom INTERFACE_LINK_LIBRARIES)
-    if(AOM_INTERFACE_LINK_LIBRARIES MATCHES vmaf)
-        # vmaf only affects the avifenc and avifdec targets below, not the avif library nor examples targets.
-        set(AVIF_USE_CXX ON)
-    endif()
 endif()
 
-if(AVIF_USE_CXX)
+if(AVIF_USE_CXX
+   OR AVIF_BUILD_APPS
```

Comment:
Optional: Right now avifgainmaputil is the only app that uses C++, and it is an experimental feature. So this line can be changed to:

```
   OR (AVIF_BUILD_APPS AND AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
```